reanalysis/comparison.py

```diff
@@ -379,24 +701,86 @@ def check_mt(
         var_mt = find_variant_in_mt(matrix=matrix, var=variant)
 
         # check if there are remaining variant rows
-        if var_mt.count_rows() == 0:
+        if not var_mt.count_rows():
             not_in_mt[sample].append(variant)
-
-        # if it is found, but was not in the VCF, it was untiered
-        else:
-            untiered[sample].append((variant, []))
+            continue
+
+        # shift attributes in the MT, and explode on gene ID
+        var_mt = prepare_mt(var_mt)
+
+        # check for any failure reasons in the QC tests
+        reasons: list[str] = []
+
+        # run all methods relating to the quality of the variant call
+        reasons.extend(run_ac_check(var_mt, config))
+        reasons.extend(run_quality_flag_check(var_mt))
+        reasons.extend(check_variant_was_normalised(var_mt))
+        reasons.extend(filter_sample_by_ab(var_mt, sample))
+        var_mt = check_gene_is_green(matrix=var_mt, green_genes=green_genes)
```

Comment:
I know this is not part of this change, but should check_gene_is_green be called filter_to_green_genes or similar?

Reply:
I think I'd like to go the other way, where the AIP method is a 'filter', and this checking version is always prefixed with 'check_', just so they're namespaced separately. This one is running a filter though, so you're right...